lib/slack/real_time/stores/store.rb

```diff
@@ -108,20 +111,29 @@ def initialize(attrs)
         # @see https://github.com/slack-ruby/slack-api-ref/blob/master/events/channel_joined.json
         on :channel_joined do |data|
           channel_id = data.channel.id
-          channel = channels[channel_id]
-          if channel
-            channel.merge!(data.channel)
+          if data.channel.is_private?
+            channel = private_channels[channel_id]
+            if channel
+              channel.merge!(data.channel)
+            else
+              private_channels[channel_id] = Models::Channel.new(data.channel)
+            end
           else
-            channels[channel_id] = Models::Channel.new(data.channel)
+            channel = public_channels[channel_id]
+            if channel
+              channel.merge!(data.channel)
+            else
+              public_channels[channel_id] = Models::Channel.new(data.channel)
+            end
           end
         end
 
         # You left a channel.
         # @see https://api.slack.com/events/channel_left
         # @see https://github.com/slack-ruby/slack-api-ref/blob/master/events/channel_left.json
         on :channel_left do |data|
-          channel = channels[data.channel]
-          channel.members.delete(self.self.id) if channel.key? :members
+          channel = public_channels[data.channel]
```

Comment:
Is this always a public channel? I'm not so sure. Same for other channel events.

Reply:
https://github.com/slack-ruby/slack-ruby-client/pull/423#discussion_r998352545